File: jwt_android/app/src/main/java/com/daeseong/jwt_android/HS256.java
package com.daeseong.jwt_android;

import android.util.Base64;
import android.util.Log;
import com.auth0.jwt.JWT;
import com.auth0.jwt.algorithms.Algorithm;
import com.auth0.jwt.interfaces.Claim;
import com.auth0.jwt.interfaces.DecodedJWT;
import java.util.HashMap;
import java.util.Map;

public class HS256 {

    private static final String TAG = HS256.class.getSimpleName();

    private String sPWD = "password1234567890";

    public String createToken(String iss, Long exp, Boolean http, String userId, String userName) {

        String token = "";

        try {
            Map<String, Object> header = new HashMap<String, Object>();
            header.put("alg", "HS256");
            header.put("typ", "JWT");

            token = JWT.create().withHeader(header)
                    .withClaim("iss", iss)
                    .withClaim("exp", exp)
                    .withClaim("https://daeseong.com/jwt", http)
                    .withClaim("userId", userId)
                    .withClaim("userName", userName)
                    .sign(Algorithm.HMAC256(sPWD));
        }catch (Exception ex){
            ex.getMessage().toString();
        }
        return token;
    }

    private String[] split(String sToken) {
        String[] result = new String[3];
        String[] split = sToken.split("\\.");
        result[0] = split[0];
        result[1] = split[1];
        result[2] = split[2];
        return result;
    }

    public void readToken(String sToken) {

        try {

            DecodedJWT jwt = JWT.decode(sToken);

            Log.e(TAG, "jwt.getToken():" + jwt.getToken());
            Log.e(TAG, "jwt.getHeader():" + jwt.getHeader());
            Log.e(TAG, "jwt.getAlgorithm():" + jwt.getAlgorithm());
            Log.e(TAG, "jwt.getPayload():" + jwt.getPayload());
            Log.e(TAG, "jwt.getSignature():" + jwt.getSignature());

            String[] split = split(jwt.getToken());
            Log.e(TAG, "split1:" + split[0]);
            Log.e(TAG, "split2:" + split[1]);
            Log.e(TAG, "split3:" + split[2]);

            Claim claim1 = jwt.getClaim("iss");
            Claim claim2 = jwt.getClaim("exp");
            Claim claim3 = jwt.getClaim("https://daeseong.com/jwt");
            Claim claim4 = jwt.getClaim("userId");
            Claim claim5 = jwt.getClaim("userName");
            Log.e(TAG, "iss:" + claim1.asString());
            Log.e(TAG, "exp:" + claim2.asLong());
            Log.e(TAG, "https://daeseong.com/jwt:" + claim3.asBoolean());
            Log.e(TAG, "userId:" + claim4.asString());
            Log.e(TAG, "userName:" + claim5.asString());

        }catch (Exception e){
            e.getMessage().toString();
        }
    }

    public String getAlgorithm(String sToken) {

        String Algorithm = "";
        try {
            DecodedJWT jwt=JWT.decode(sToken);
            Algorithm = jwt.getAlgorithm();
        }catch (Exception e){
            e.getMessage().toString();
        }
        return Algorithm;
    }

    public String getHeader(String sToken) {

        String Header = "";
        try {
            DecodedJWT jwt=JWT.decode(sToken);
            Header = new String(Base64.decode(jwt.getHeader(), Base64.URL_SAFE));
        }catch (Exception e){
            e.getMessage().toString();
        }
        return Header;
    }

    public String getPayload(String sToken) {

        String payload = "";
        try {
            DecodedJWT jwt = JWT.decode(sToken);
            payload = new String(Base64.decode(jwt.getPayload(), Base64.URL_SAFE));
        }catch (Exception e){
            e.getMessage().toString();
        }
        return payload;
    }

    public String getSignature(String sToken) {

        String signature = "";
        try {
            DecodedJWT jwt = JWT.decode(sToken);
            signature = jwt.getSignature();
        }catch (Exception e){
            e.getMessage().toString();
        }
        return signature;
    }
}

File: jwt_android(kotlin)/app/src/main/java/com/daeseong/jwt_android/HS256.kt
package com.daeseong.jwt_android

import android.util.Base64
import android.util.Log
import com.auth0.jwt.JWT
import com.auth0.jwt.algorithms.Algorithm

class HS256 {

    private val tag = HS256::class.java.simpleName
    private val sPWD = "password1234567890"

    fun createToken(iss: String?, exp: Long?, http: Boolean?, userId: String?, userName: String?): String {
        return try {
            val header: Map<String, Any> = mapOf("alg" to "HS256", "typ" to "JWT")
            JWT.create().withHeader(header)
                .withClaim("iss", iss)
                .withClaim("exp", exp)
                .withClaim("https://daeseong.com/jwt", http)
                .withClaim("userId", userId)
                .withClaim("userName", userName)
                .sign(Algorithm.HMAC256(sPWD))
        } catch (ex: Exception) {
            Log.e(tag, ex.message.toString())
            ""
        }
    }

    private fun split(sToken: String): Array<String?> {
        return sToken.split("\\.").toTypedArray()
    }

    fun readToken(sToken: String?) {
        try {
            val jwt = JWT.decode(sToken)
            Log.e(tag, "jwt.getToken(): ${jwt.token}")
            Log.e(tag, "jwt.getHeader(): ${jwt.header}")
            Log.e(tag, "jwt.getAlgorithm(): ${jwt.algorithm}")
            Log.e(tag, "jwt.getPayload(): ${jwt.payload}")
            Log.e(tag, "jwt.getSignature(): ${jwt.signature}")

            val split = split(jwt.token)
            Log.e(tag, "split1: ${split[0]}")
            Log.e(tag, "split2: ${split[1]}")
            Log.e(tag, "split3: ${split[2]}")

            val claim1 = jwt.getClaim("iss")
            val claim2 = jwt.getClaim("exp")
            val claim3 = jwt.getClaim("https://daeseong.com/jwt")
            val claim4 = jwt.getClaim("userId")
            val claim5 = jwt.getClaim("userName")

            Log.e(tag, "iss: ${claim1.asString()}")
            Log.e(tag, "exp: ${claim2.asLong()}")
            Log.e(tag, "https://daeseong.com/jwt: ${claim3.asBoolean()}")
            Log.e(tag, "userId: ${claim4.asString()}")
            Log.e(tag, "userName: ${claim5.asString()}")

        } catch (e: Exception) {
            Log.e(tag, e.message.toString())
        }
    }

    fun getAlgorithm(sToken: String?): String {
        return try {
            val jwt = JWT.decode(sToken)
            jwt.algorithm
        } catch (e: Exception) {
            Log.e(tag, e.message.toString())
            ""
        }
    }

    fun getHeader(sToken: String?): String {
        return try {
            val jwt = JWT.decode(sToken)
            String(Base64.decode(jwt.header, Base64.URL_SAFE))
        } catch (e: Exception) {
            Log.e(tag, e.message.toString())
            ""
        }
    }

    fun getPayload(sToken: String?): String {
        return try {
            val jwt = JWT.decode(sToken)
            String(Base64.decode(jwt.payload, Base64.URL_SAFE))
        } catch (e: Exception) {
            Log.e(tag, e.message.toString())
            ""
        }
    }

    fun getSignature(sToken: String?): String {
        return try {
            val jwt = JWT.decode(sToken)
            jwt.signature
        } catch (e: Exception) {
            Log.e(tag, e.message.toString())
            ""
        }
    }
}
